pkg/controller/kieapp/kieapp_controller.go

```diff
@@ -351,6 +353,29 @@ func (reconciler *Reconciler) dcUpdateCheck(current, new oappsv1.DeploymentConfi
 	return dcUpdates
 }
 
+func (reconciler *Reconciler) hasUpdatedVolumes(current, new corev1.PodSpec) bool {
+	cContainer := current.Containers[0]
+	nContainer := new.Containers[0]
+
+	if len(current.Volumes) != len(new.Volumes) || len(cContainer.VolumeMounts) != len(nContainer.VolumeMounts) {
+		return true
+	}
+
+	for i := range current.Volumes {
+		if !strings.HasSuffix(current.Volumes[i].Name, constants.KeystoreVolumeSuffix) && !reflect.DeepEqual(current.Volumes[i], new.Volumes[i]) {
```

Comment:
can we expect that the ordering of old and new volume/mount arrays will be identical? should we sort them by name first to ensure proper comparison? i'm not aware of whether ocp does any ordering post deployment.

Reply:
Removed this logic